src/lib/EventPublisher.js

```diff
@@ -82,7 +82,11 @@ class EventPublisher extends EventEmitter {
     filterOptions?: FilterOptions = {},
     subscriberID?: string,
   ): void => {
-    const subscriptionID = uuid();
+    let subscriptionID = uuid();
+    while(this._subscriptionsByID.has(subscriptionID)) {
```

Comment:
You actually don't need this here.  The subscriptions are stored in memory so there should never be any collisions.  The only time a collision will happen is when we are loading IDs from disk.